source/signalzeug/source/ScopedConnection.cpp

```diff
@@ -14,15 +14,32 @@ ScopedConnection::ScopedConnection(const Connection & connection)
 {
 }
 
+ScopedConnection::ScopedConnection(ScopedConnection && other)
+:   m_connection(other.m_connection)
+{
+    other.m_connection = Connection();
+}
+
 ScopedConnection::~ScopedConnection()
 {
 	m_connection.disconnect();
 }
 
-void ScopedConnection::operator=(const Connection & connection)
+ScopedConnection& ScopedConnection::operator=(const Connection & connection)
 {
 	m_connection.disconnect();
 	m_connection = connection;
+
+    return *this;
+}
+
+ScopedConnection& ScopedConnection::operator=(ScopedConnection && other)
+{
+    m_connection.disconnect();
+    m_connection = other.m_connection;
```

Comment:
The passed parameter is a r-value reference, indicating that the memory behind it will not be used further. So why bother with filling valid data in ```other.m_connection```?
The one thing that is important is the movement of the value, not a swap.

Or does I misunderstand those r-value references?